frontend/src/components/stages/base_stage_editor.ts

```diff
@@ -129,16 +129,90 @@ export class BaseStageEditorComponent extends MobxLitElement {
     `;
   }
 
+  private shouldDisableWaitForAllParticipants(): boolean {
+    if (!this.stage || this.stage.kind !== StageKind.REVEAL) {
+      return false;
+    }
+
+    for (const item of this.stage.items) {
+      // There's a dependency on all participants if we want to reveal all results.
+      if (
+        'revealAudience' in item &&
+        item.revealAudience === RevealAudience.ALL_PARTICIPANTS
+      ) {
+        return true;
+      }
+      if (item.kind === StageKind.RANKING) {
+        const stageId = item.id;
+
+        const foundStage = this.experimentEditor.stages.find(
+          (stage) => stage.id === stageId
+        );
+        // There's a dependency on all participants if we're ranking all participants.
+        if (
+          foundStage &&
+          'rankingType' in foundStage &&
+          foundStage.rankingType === RankingType.PARTICIPANTS
+        ) {
+          return true;
+        }
+
+        // There's a dependency on all participants if there's an election
+        // (so all votes are counted).
+        if (
+          foundStage &&
+          'strategy' in foundStage &&
+          foundStage.strategy === ElectionStrategy.CONDORCET
+        ) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
   private renderWaitForAllParticipants() {
     if (!this.stage) return nothing;
     const waitForAllParticipants = this.stage.progress.waitForAllParticipants;
 
     const updateCheck = () => {
       if (!this.stage) return;
-      const progress: StageProgressConfig = { ...this.stage.progress, waitForAllParticipants: !waitForAllParticipants };
-      this.experimentEditor.updateStage({ ...this.stage, progress });
+      const progress: StageProgressConfig = {
+        ...this.stage.progress,
+        waitForAllParticipants: !waitForAllParticipants,
+      };
+      this.experimentEditor.updateStage({...this.stage, progress});
     };
 
+    if (this.shouldDisableWaitForAllParticipants()) {
```

Comment:
Instead of rendering twice (in this if statement and below), can we just have the main statement and conditionally set render the warning, set disabled, etc.?